.github/workflows/ci.yaml

```diff
@@ -0,0 +1,188 @@
+          # since it has huge performance overhead (tests are 50% or so slower)
+          ENABLE_COVERAGE=$([ "${GITHUB_EVENT_NAME}" != "pull_request" ] && [ "${IS_NIGHTLY_BUILD}" = "no" ] && echo "yes" || echo "no")
+          echo "ENABLE_COVERAGE=${ENABLE_COVERAGE}" >> $GITHUB_ENV
+
+          # We only run tests with "--with-timer" flag on master and not for PRs since it adds 1-2
+          # minutes of overhead to each build.
+          NOSE_TIME=$([ "${GITHUB_EVENT_NAME}" != "pull_request" ] && [ "${IS_NIGHTLY_BUILD}" = "no" ] && echo "yes" || echo "no")
+          echo "NOSE_TIME=${NOSE_TIME}" >> $GITHUB_ENV
+
+          # Setup the path to the st2 repo in the CI build system
+          echo "ST2_CI_REPO_PATH=${GITHUB_WORKSPACE}" >> $GITHUB_ENV
+      - name: Checkout repository
+        uses: actions/checkout@v2
+      - name: 'Set up Python (${{ matrix.python-version }})'
+        uses: actions/setup-python@v2
+        with:
+          python-version: '${{ matrix.python-version }}'
+      - uses: actions/cache@v2
+        with:
+          path: |
+            .cache/pip
+            virtualenv
+          key: ${{ runner.os }}-${{ matrix.python-version }}-${{ hashFiles('requirements.txt', 'test-requirements.txt') }}
+          restore-keys: |
+            ${{ runner.os }}-${{ matrix.python }}-
+      - name: Install apt depedencies
+        run: |
+          # install dev dependencies for Python LDAP module
+          # https://github.com/StackStorm/st2-auth-ldap
+          sudo apt-get -y update
+          sudo apt-get -f -y install libldap2-dev libsasl2-dev libssl-dev ldap-utils
+      - name: Install virtualenv
+        run: |
+          # Note: Use the verison of virtualenv pinned in fixed-requirements.txt so we
+          #       only have to update it one place when we change the version
+          pip install --upgrade --force-reinstall $(grep "^virtualenv" fixed-requirements.txt)
+      - name: Install requirements
+        run: |
+          ./scripts/travis/install-requirements.sh
+      - name: Setup integration tests
+        run: |
+          # prep a ci-specific dev conf file that uses runner instead of stanley
+          # this user is the username of the user in GitHub actions, used for SSH, etc during
+          # integration tests (important)
+          cp conf/st2.dev.conf "${ST2_CONF}" ; sed -i -e "s/stanley/${ST2_CI_USER}/" "${ST2_CONF}"
+          scripts/travis/add-itest-user-key.sh
+          sudo .circle/add-itest-user.sh
+      - name: Permissions Workaround
+        if: "${{ env.TASK == 'ci-packs-tests' || env.TASK == 'ci-integration' }}"
+        run: |
+          echo "$ST2_CI_REPO_PATH"
+          sudo ST2_CI_REPO_PATH="${ST2_CI_REPO_PATH}" scripts/travis/permissions-workaround.sh
+      - name: Setup RabbitMQ (NOT WORKING YET)
+        if: "${{ env.TASK == 'ci-integration' }}"
+        run: |
+          # Use custom RabbitMQ config which enables SSL / TLS listener on port 5671 with test certs
```

Comment:
I think eventually it would be nice to move this rabbitmq stuff inside a service Docker container which runs in the background for the duration of the build.

Reply:
EDIT: Never mind, I see the comment here now - https://github.com/StackStorm/st2/pull/5098/files#diff-944291df2c9c06359d37cc8833d182d705c9e8c3108e7cfe132d61a06e9133ddR42.